remotespark/RemoteSparkMagics.py

```diff
@@ -125,40 +123,35 @@ def spark(self, line, cell=""):
         # run
         elif len(subcommand) == 0:
             if args.context == Constants.context_name_spark:
-                (success, out) = self.spark_controller.run_cell(cell,
-                                                                args.session)
+                (success, out) = self.spark_controller.run_cell(cell, args.session)
                 if success:
-                    self.ipython.write(out)
+                    self.shell.write(out)
```

Comment:
Yes, I did verify.

________________________________
From: msftristew <notifications@github.com>
Sent: Friday, December 18, 2015 9:57:16 AM
To: jupyter-incubator/sparkmagic
Cc: Alejandro Guerrero Gonzalez
Subject: Re: [sparkmagic] Add df to user context when executing SQL script when requested via -o (#74)


In remotespark/RemoteSparkMagics.py<https://github.com/jupyter-incubator/sparkmagic/pull/74#discussion_r48051165>:

>                  if success:
> -                    self.ipython.write(out)
> +                    self.shell.write(out)


Did you verify that shell will always point to an InteractiveShell instance? Otherwise this will crash at runtime.

get_ipython() never returns None from what I can tell so I don't see the reason to move away from it.

-
Reply to this email directly or view it on GitHub<https://github.com/jupyter-incubator/sparkmagic/pull/74/files#r48051165>.